### PR TITLE
Bugfix: disabled check added to button component

### PR DESCRIPTION
**Pull request by verena-ifx**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
The button handleClick method didnt have a disabled check. Added to prevent it from being clickable when disabled

<details>
  <summary>📦 Published PR as canary version: <code>20.36.2--canary.794.814805cda3dabb0eb9b01c9ca851c907691f7155.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @infineon/infineon-design-system-stencil@20.36.2--canary.794.814805cda3dabb0eb9b01c9ca851c907691f7155.0
  # or 
  yarn add @infineon/infineon-design-system-stencil@20.36.2--canary.794.814805cda3dabb0eb9b01c9ca851c907691f7155.0
  ```
</details>

### CODE REVIEW

**Comment by tishoyanchev** [comment]
General issue:
- The focus state does not work. To fix it, simply add **tabindex={1}** on the anchor.

File: examples/wrapper-components/vue-javascript/src/assets/main.scss [hunk 1]
```diff
@@ -1 +1 @@
-@import './base.scss';
\ No newline at end of file
+// @import './base.scss';
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply examples/wrapper-components/vue-javascript/src/assets/main.scss:1]: the thing is, its for demonstration of how to use the tokens. But I noticed, that when using symlinks to run the package locally, the tokens project doesnt get imported (symlinks cant link more than 1 package at once).
So it works perfeclty fine when using it in a real app, but not with "npm run test:local"
But you are right, maybe we should remove it and demonstrate the usage in separate example repositories or our templates. Let me think about this for 1-2 days and then discuss? (You can of course also think about it in the meantime)

File: examples/wrapper-components/vue-javascript/src/components/Button.vue [hunk 2]
```diff
@@ -4,8 +4,7 @@
   <div>
     <h2>Button</h2>
     <div>
-      <ifx-button theme='default' @click="handleClick"
-        @mouseover="handleMouseOver">
+      <ifx-button theme='default' @click="handleClick" disabled="true" @mouseover="handleMouseOver">
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply examples/wrapper-components/vue-javascript/src/components/Button.vue:7]: Yes, will do! (+ in the other example apps)

File: packages/components/src/components/button/button.tsx [hunk 3]
```diff
@@ -91,13 +93,6 @@ export class Button {
...
 
-  // handleFocus(event: FocusEvent) { // the anchor element should not be focusable when it's disabled
-  //   if (this.disabled) {
-  //     event.preventDefault();
-  //     this.focusableElement.blur();
-  //   }
-  // }
-
 
   handleFocus(event: FocusEvent) {
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply packages/components/src/components/button/button.tsx:97]: Hm i havent touched the focus state of the button in my changes. But yes, lets fix it in this PR as well then

File: examples/wrapper-components/vue-javascript/src/assets/main.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (verena-ifx) [reply examples/wrapper-components/vue-javascript/src/assets/main.scss:1]: Ok maybe I found a solution, a quite simple one actually

**Comment by tishoyanchev** [comment]
> General issue:
> 
> * The focus state does not work. To fix it, simply add **tabindex={1}** on the anchor.
> 
> I can't mark down the code line with the review tool, so I'll write it here. The reason the focus does not work is because: Inside here: **private addEventListenersToHandleCustomFocusAndActiveState()** **const element = this.el.shadowRoot.firstChild;** on this line, the **firstChild** is the **<style>** tag, and not the . So the **tabIndex** attribute is added to the **<style>**. I also believe the **tabIndex = 0** should be **1** instead.

**Outdated!**

**Pull request merged**